numpy/random/mtrand/mtrand.pyx

```diff
@@ -1551,18 +1551,17 @@ cdef class RandomState:
         cdef double flow, fhigh, fscale
         cdef object temp
 
-        flow = PyFloat_AsDouble(low)
-        fhigh = PyFloat_AsDouble(high)
+        if np.array(low).shape == np.array(high).shape == ():
```

Comment:
It seems to have worked.  I tested the methods briefly by hand (more formal testing is coming though!), and I didn't have any issues with the double (or even triple) comparison with regards to output data type.